modules/control/control_component.cc

```diff
@@ -305,13 +311,27 @@ bool ControlComponent::Proc() {
     OnPad(pad_msg);
   }
 
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Add TODO here for later refactor to avoid redundant copies.

Reply:
Added. Thank you.